exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/CryptoFunctions.java

```diff
@@ -0,0 +1,389 @@
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Override
+    public void setup() {
+
+    }
+
+    @Override
+    public void eval() {
+
+      String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawInput.start, rawInput.end, rawInput.buffer);
+
+      String sha384 = org.apache.commons.codec.digest.DigestUtils.sha384Hex(input);
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = sha384.getBytes().length;
+      buffer.setBytes(0, sha384.getBytes());
+    }
+
+  }
+
+  /**
+   * This function returns the SHA512 digest of a given input string.
+   *  Usage is shown below:
+   *  select sha512( 'testing' ) from (VALUES(1));
+   */
+
+
+  @FunctionTemplate(name = "sha512", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class SHA512Function implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Override
+    public void setup() {
+
+    }
+
+    @Override
+    public void eval() {
+
+      String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawInput.start, rawInput.end, rawInput.buffer);
+
+      String sha512 = org.apache.commons.codec.digest.DigestUtils.sha512Hex(input);
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = sha512.getBytes().length;
+      buffer.setBytes(0, sha512.getBytes());
+    }
+
+  }
+
+  /**
+   * aes_encrypt()/ aes_decrypt(): implement encryption and decryption of data using the official AES (Advanced Encryption Standard) algorithm,
+   * previously known as “Rijndael.” AES_ENCRYPT() encrypts the string str using the key string key_str and returns a
+   * binary string containing the encrypted output.
+   * Usage:  SELECT aes_encrypt( 'encrypted_text', 'my_secret_key' ) AS aes FROM (VALUES(1));
+   */
+
+
+  @FunctionTemplate(name = "aes_encrypt", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class AESEncryptFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Param
+    VarCharHolder rawKey;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Workspace
+    Cipher cipher;
+
+    @Override
+    public void setup() {
+      String key = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawKey.start, rawKey.end, rawKey.buffer);
+      java.security.MessageDigest sha = null;
+      try {
+        byte[] keyByteArray = key.getBytes("UTF-8");
+        sha = java.security.MessageDigest.getInstance("SHA-1");
+        keyByteArray = sha.digest(keyByteArray);
+        keyByteArray = java.util.Arrays.copyOf(keyByteArray, 16);
+        SecretKeySpec secretKey = new SecretKeySpec(keyByteArray, "AES");
```

Comment:
We don't need to use fully qualified class name when variable is declared in class (like for `Cipher`, import will be added automatically). For `SecretKeySpec` we need to use fully qualified class name since it's declared internally in method: `javax.crypto.spec.SecretKeySpec secretKey = new javax.crypto.spec.SecretKeySpec(keyByteArray, "AES");`

For example, Drill will create `ProjectorGen0` where it will add variables declared in class (thus imports will be added automatically). Also it will copy content of `setup` and `eval` methods, in this case imports won't be added automatically. We might need to enhance this mechanism so all imports UDFs class are added and user won't required to use fully qualified class names. The same problem be with `logger`.

Actually, I have run `TestCryptoFunctions` on your branch and `testAESEncrypt()` and `testAESDecrypt()` fail.
First with ` java.lang.Exception: org.apache.drill.common.exceptions.UserRemoteException: SYSTEM ERROR: CompileException: Line 55, Column 32: Cannot determine simple type name "SecretKeySpec"` and when I have fixed this problem with `Caused By (org.codehaus.commons.compiler.CompileException) Line 61, Column 16: Unknown variable or type "logger"`.

Note: it's a good practice (I even would say a requirement) to make sure that all unit test pass before publishing changes in Drill.